File: CS/19_OLE/ExtractOLEObject.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using System.IO;

namespace ExtractOLEObject
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PPT document
            Presentation presentation = new Presentation();

            //Load document from disk
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\ExtractOLEObject.pptx");

            //Loop through the slides and shapes
            foreach (ISlide slide in presentation.Slides)
            {
                foreach (IShape shape in slide.Shapes)
                {
                    if (shape is IOleObject)
                    {
                        //Find OLE object
                        IOleObject oleObject = shape as IOleObject;

                        //Get its data and write to file
                        byte[] bytes = oleObject.Data;
                        switch (oleObject.ProgId)
                        {
                            case "Excel.Sheet.8":
                                File.WriteAllBytes("result.xls", bytes);
                                break;
                            case "Excel.Sheet.12":
                                File.WriteAllBytes("result.xlsx", bytes);
                                break;
                            case "Word.Document.8":
                                File.WriteAllBytes("result.doc", bytes);
                                break;
                            case "Word.Document.12":
                                File.WriteAllBytes("result.docx", bytes);
                                break;
                            case "PowerPoint.Show.8":
                                File.WriteAllBytes("result.ppt", bytes);
                                break;
                            case "PowerPoint.Show.12":
                                File.WriteAllBytes("result.pptx", bytes);
                                break;
                        }
                    }
                }
            }
            MessageBox.Show("Completed!");
        }
    }
}

File: VB/19_OLE/ExtractOLEObject.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation
Imports System.IO

Namespace ExtractOLEObject
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PPT document
			Dim presentation As New Presentation()

			'Load document from disk
			presentation.LoadFromFile("..\..\..\..\..\..\Data\ExtractOLEObject.pptx")

			'Loop through the slides and shapes
			For Each slide As ISlide In presentation.Slides
				For Each shape As IShape In slide.Shapes
					If TypeOf shape Is IOleObject Then
						'Find OLE object
						Dim oleObject As IOleObject = TryCast(shape, IOleObject)

						'Get its data and write to file
						Dim bytes() As Byte = oleObject.Data
						Select Case oleObject.ProgId
							Case "Excel.Sheet.8"
								File.WriteAllBytes("result.xls", bytes)
							Case "Excel.Sheet.12"
								File.WriteAllBytes("result.xlsx", bytes)
							Case "Word.Document.8"
								File.WriteAllBytes("result.doc", bytes)
							Case "Word.Document.12"
								File.WriteAllBytes("result.docx", bytes)
							Case "PowerPoint.Show.8"
								File.WriteAllBytes("result.ppt", bytes)
							Case "PowerPoint.Show.12"
								File.WriteAllBytes("result.pptx", bytes)
						End Select
					End If
				Next shape
			Next slide
			MessageBox.Show("Completed!")
		End Sub
	End Class
End Namespace